src/basic/strxcpyx.c

```diff
@@ -15,89 +15,116 @@
 #include <stdio.h>
 #include <string.h>
 
+#include "string-util.h"
 #include "strxcpyx.h"
 
-size_t strnpcpy(char **dest, size_t size, const char *src, size_t len) {
+size_t strnpcpy_full(char **dest, size_t size, const char *src, size_t len, bool *ret_truncated) {
+        bool truncated = false;
```

Comment:
Please add some tests for all the new functions. Those things are usually tricky to get right in all corner cases.